lib/aptible/cli/helpers/s3_log_helpers.rb

```diff
@@ -0,0 +1,218 @@
+            properties[:container_id] = split_by_dot.first.delete!('-json')
+            properties[:start_time] = utc_datetime(split_by_dot[-2])
+            properties[:end_time] = utc_datetime(split_by_dot[-1])
+          else
+            m = "Cannot determine aptible log naming schema from #{file}"
+            raise Thor::Error, m
+          end
+          properties
+        end
+
+        def decrypt_and_translate_s3_file(file, enc_key, region, bucket, path)
+          # AWS warns us about using the legacy encryption schema
+          s3 = Kernel.silence_warnings do
+            Aws::S3::EncryptionV2::Client.new(
+              encryption_key: enc_key, region: region,
+              key_wrap_schema: :aes_gcm,
+              content_encryption_schema: :aes_gcm_no_padding,
+              security_profile: :v2_and_legacy
+            )
+          end
+
+          # Just write it to a file directly
+          location = File.join(path, file.split('/').drop(4).join('/'))
+          FileUtils.mkdir_p(File.dirname(location))
+          File.open(location, 'wb') do |f|
+            CLI.logger.info location
+            # Is this memory efficient?
+            s3.get_object(bucket: bucket, key: file, response_target: f)
+          end
+        end
+
+        def find_s3_files_by_string_match(region, bucket, stack, strings)
+          # This function just regex matches a provided string anywhwere
+          # in the s3 path
+          begin
+            stack_logs = s3_client(region).bucket(bucket)
+                                          .objects(prefix: stack)
+                                          .map(&:key)
+          rescue => error
+            raise Thor::Error, error.message
+          end
+          strings.each do |s|
+            stack_logs = stack_logs.select { |f| f =~ /#{s}/ }
+          end
+          stack_logs
+        end
+
+        def find_s3_files_by_attrs(region, bucket, stack,
+                                   attrs, time_range = nil)
+          # This function uses the known path schema to return files matching
+          # any provided criteria. EG:
+          # * attrs: { :type => 'app', :id => 123 }
+          # * attrs: { :container_id => 'deadbeef' }
+
+          begin
+            stack_logs = s3_client(region).bucket(bucket)
+                                          .objects(prefix: stack)
+                                          .map(&:key)
+          rescue => error
+            raise Thor::Error, error.message
+          end
+          attrs.each do |k, v|
+            stack_logs = stack_logs.select do |f|
+              if k == :container_id
+                # Match short container IDs
+                info_from_path(f)[k].start_with?(v)
+              else
+                info_from_path(f)[k] == v
+              end
+            end
+          end
+
+          if time_range
+            # select only logs within the time range
+            # TODO handle 'unknown' time ranges
```

Comment:
Is this still TODO since `rescue ArgumentError` was added to the `utc_datetime` method?

Reply:
No.